mirrord/operator/src/client.rs

```diff
@@ -1288,8 +1290,24 @@ impl OperatorApi<PreparedClientCert> {
             list_reusable_mysql_branches(&mysql_branch_api, &create_mysql_params, &subtask).await?;
 
         create_mysql_params.retain(|id, _| !reusable_mysql_branches.contains_key(id));
+
+        // Get the maximum timeout from all DB branch configs
+        let timeout_secs = layer_config
+            .feature
+            .db_branches
+            .iter()
+            .map(|branch_config| match branch_config {
+                mirrord_config::feature::database_branches::DatabaseBranchConfig::Mysql(
+                    mysql_config,
+                ) => mysql_config.base.creation_timeout_secs,
+            })
+            .max()
```

Comment:
Hmm, I was thinking to let each branch's `await_condition` wrapped by its own timeout since it's configured separately. But this makes sense too and simplifies the logic.